src/dearpygui_commands.h

```diff
@@ -2125,7 +2127,7 @@ split_frame(PyObject* self, PyObject* args, PyObject* kwargs)
 		&delay))
 		return GetPyNone();
 
-	if (!GContext->manualMutexControl) std::lock_guard<std::mutex> lk(GContext->mutex);
+	std::lock_guard<std::recursive_mutex> lk(GContext->mutex);
```

Comment:
I've just found a problem with this fix: it breaks `split_frame()`. Since it now properly locks the mutex, there's very little chance that `waitOneFrame` goes back from `true` to `false`. To fix that, it's sufficient to remove the line 2130 (locking the mutex) altogether: `split_frame` doesn't need the mutex.

Reply:
Also, while `split_frame` can be fixed by removing the mutex lock in it, it could also be made more efficient using a condition variable instead of an atomic bool. This way it returns as soon as rendering of the new frame has ended, instead of polling the variable every 32 ms (and potentially waiting up to 32 ms longer than needed).

I've implemented this in my local copy and can share as a separate PR.

Reply:
You can submit with this PR if you'd like.